### PR TITLE
WRR-3893: Updated node modules' major versions

### PR DESCRIPTION
**Pull request by paul-beldean-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Updated node modules' major versions:
electron  ^31.2.1  →  ^32.1.0
@types/node  ^20.14.8  →  ^22.5.5

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Updated node modules' major versions

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
Updated node engine version for electron to v20.17.0.
Updated electron version in browserslist

### Links
[//]: # (Related issues, references)
WRR-3893

### Comments

Enact-DCO-1.0-Signed-off-by: Paul Beldean paul.beldean@lgepartner.com

### CODE REVIEW

**Review by daniel-stoian-lgp — APPROVED** [review]

File: packages/electron/template/package.json [hunk 1]
```diff
@@ -25,15 +25,15 @@
   },
   "license": "UNLICENSED",
   "engines": {
-    "node": ">=20.14.0"
+    "node": ">=20.17.0"
```
Comment (daniel-stoian-lgp) [inline packages/electron/template/package.json:28]: Note for the next reviewer: 
I think during the last major dependency update, we confused "stack upgrade" with "minimum node version".
https://github.com/electron/electron/releases/tag/v32.0.0 
According to npm (check package.json), minimum node version is 12: https://www.npmjs.com/package/electron?activeTab=code 
image

Should we fix or remove the minimum supported node version? Or align it with Enact/cli?

Comment (hong6316) [reply packages/electron/template/package.json:28]: Please check again, node v20 also described in `DEPS`
https://github.com/electron/electron/blob/5a97ee26ff0ce322ddd454cc9aee864f7898258c/DEPS#L6

We need more information about the minimum node version...

Comment (paul-beldean-lgp) [reply packages/electron/template/package.json:28]: Taeyoung, I think you are right that it should be v20, I think it was a mistake that they let v12 in package.json.
In package.json for the last release there is no longer a node engine version and also the version for `@types/node` is set to `^20.9.0`
https://github.com/electron/electron/blob/v32.1.2/package.json

**Review by hong6316 — APPROVED** [review]
👍

**Pull request merged**